src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -87,6 +88,10 @@ public static function getSubscribedEvents()
 
     private function shouldRetry(\Throwable $e, Envelope $envelope, RetryStrategyInterface $retryStrategy): bool
     {
+        if ($e instanceof RecoverableExceptionInterface) {
+            return true;
+        }
```

Comment:
What about the `HandlerFailedException` case where you need to check for the nested exceptions? 

Also, a new test case for the listener would be ideal :)